nautobot_circuit_maintenance/views.py

```diff
@@ -299,3 +314,76 @@ def get(self, request, *args, **kwargs):
                 "source_type": source.__class__.__name__,
             },
         )
+
+
+def google_authorize(request, slug):
+    """View to start the Google OAuth authorization flow."""
+    # Create flow instance to manage the OAuth 2.0 Authorization Grant Flow steps.
+    notification_source = models.NotificationSource.objects.get(slug=slug)
+    source = Source.init(name=notification_source.name)
+    request.session["CLIENT_SECRETS_FILE"] = source.credentials_file
+    request.session["SCOPES"] = source.SCOPES
+    request.session["SOURCE_SLUG"] = slug
+
+    # Create flow instance to manage the OAuth 2.0 Authorization Grant Flow steps.
+    flow = google_auth_oauthlib.flow.Flow.from_client_secrets_file(
+        request.session["CLIENT_SECRETS_FILE"], scopes=request.session["SCOPES"]
+    )
+    # The URI created here must exactly match one of the authorized redirect URIs
+    # for the OAuth 2.0 client, which you configured in the API Console. If this
+    # value doesn't match an authorized URI, you will get a 'redirect_uri_mismatch'
+    # error.
+
+    flow.redirect_uri = (
+        request.scheme
+        + "://"
+        + request.get_host()
+        + reverse("plugins:nautobot_circuit_maintenance:google_oauth2callback")
+    )
+    authorization_url, state = flow.authorization_url(
+        # Enable offline access so that you can refresh an access token without
+        # re-prompting the user for permission. Recommended for web server apps.
+        access_type="offline",
+        # Enable incremental authorization. Recommended as a best practice.
+        include_granted_scopes="true",
+        # Asking for consent everytime is providing a refresh token that if not used only happens
+        # the first time.
```

Comment:
I'm having trouble parsing this sentence. Can you perhaps rephrase?

Reply:
hopefully better now :)